text/0016-network-spawn.md

```diff
@@ -0,0 +1,76 @@
+- Feature Name: network-spawn
+- Start Date: 2021-05-25
+- RFC PR: [Unity-Technologies/com.unity.multiplayer.rfcs#0016](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/pull/0016)
+- Issue: [Unity-Technologies/com.unity.multiplayer#0000](https://github.com/Unity-Technologies/com.unity.multiplayer/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Rename `NetworkStart` to `OnNetworkSpawn` and introduce `OnNetworkDespawn` function.
+
+# Motivation
+[motivation]: #motivation
+
+- Many users try to subscribe to events in `NetworkStart` but then have no place to unsubscribe. As a user I want to know when my NetworkObject gets spawned and despawned. The proposed API will make this accessible.
+- `NetworkStart` might lead people to think that it is a similar initialization function as `Start` or `OnEnable` which is not true. The new wording will be more direct.
+- This will also make pooling/addressable support etc. much more straight forward because we decouple object life cycle from network lifecycle.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+`NetworkStart` has been renamed to `OnNetworkSpawn`. `OnNetworkSpawn` gets called when a server spawns an object on the server and gets called when a client replicates the spawn of a NetworkObject.
+
+`OnNetworkDespawn` gets called when the NetworkObject gets despawned. On the server this will be called when the object gets despawned with `NetworkObject::Despawn` or destroyed. On the client this will be called when a NetworkObject gets despawned by the server.
+
+`OnNetworkSpawn` and `OnNetworkDepawn` are great places to subscribe to network related events such as OnValueChanged from NetworkVariable.
+```csharp
+public class Example : NetworkBehaviour
+{
+    private NetworkVariableFloat m_Variable = new NetworkVariableFloat();
+
+    public override void OnNetworkSpawn()
+    {
+        m_Variable.OnValueChanged += MyValueChangeFunction;
+    }
+
+    public override void OnNetworkDespawn()
+    {
+        m_Variable.OnValueChanged -= MyValueChangeFunction;
+    }
+
+    private void MyValueChangeFunction(float prev, float next)
+    {
+        Debug.Log($"previous {prev}, next {next}");
+    }
+}
+```
+
+# Reference-level explanation
```

Comment:
makes a lot of sense, thank you for this RFC.
I only have one question, do you mind outlining where/how you'd wire `OnNetworkDespawn` events on the server and clients?
Some neighbor codes and/or call-stack to get to the `OnNetworkDespawn` would be great!

Reply:
I added this to the reference level section of the RFC. I also plan to do some small refactoring around how NetworkStart is called (no behaviour changes just throwing out unused legacy code)